Patient Registry System/PatientRegistrySystem.DB/Repos/DbGenericRepository/DbGeneric.cs

```diff
@@ -0,0 +1,249 @@
+﻿using AutoMapper;
+using Microsoft.EntityFrameworkCore;
+using PatientRegistrySystem.DB.Contexts;
+using PatientRegistrySystem.DB.Models;
+using PatientRegistrySystem.DB.Models.DbModels;
+using PatientRegistrySystem.Domain.Dto;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace PatientRegistrySystem.DB.Repos.DbGenericRepository
+{
+    public class DbGeneric<DbEntity, DomainEntity> : IDbGeneric<DbEntity, DomainEntity> where DbEntity : class, IDbModel where DomainEntity : IDomainModel
+    {
+        private readonly ApplicationIdentityDbContext _applicationIdentityDbContext;
+        private readonly IMapper _mapper;
+        private DbSet<DbEntity> _entities;
+
+        public DbGeneric(ApplicationIdentityDbContext applicationIdentityDbContext, IMapper mapper)
+        {
+            _entities = applicationIdentityDbContext.Set<DbEntity>();
+            _applicationIdentityDbContext = applicationIdentityDbContext;
+            _mapper = mapper;
+        }
+
+        public async Task<bool> CreateEntityAsync(DomainEntity entity)
+        {
+            if (entity.GetType().Equals(typeof(Record))
+                || entity.GetType().Equals(typeof(Prescription))
+                || entity.GetType().Equals(typeof(Company))
+                || entity.GetType().Equals(typeof(Medicine)))
+            {
+                await _entities.AddAsync(_mapper.Map<DbEntity>(entity));
+                return await SaveChangesAsync();
+            }
+            else
+            {
+                return false;
+            }
+        }
+
+        public async Task<bool> DeleteEntityDeepAsync(DomainEntity entity)
+        {
+            if (entity.GetType().Equals(typeof(Record))
+                || entity.GetType().Equals(typeof(Prescription))
+                || entity.GetType().Equals(typeof(Company))
+                || entity.GetType().Equals(typeof(Medicine)))
+            {
+                _entities.Remove(_mapper.Map<DbEntity>(entity));
+                return await SaveChangesAsync();
+            }
+            else
+            {
+                return false;
+            }
+        }
+
+        public async Task<bool> DeleteEntitySoftAsync(DomainEntity entity)
+        {
+            if (entity.GetType().Equals(typeof(Record))
+                || entity.GetType().Equals(typeof(Prescription))
+                || entity.GetType().Equals(typeof(Company))
+                || entity.GetType().Equals(typeof(Medicine)))
+            {
+                _entities.Update(_mapper.Map<DbEntity>(entity));
+                return await SaveChangesAsync();
+            }
+            else
+            {
+                return false;
+            }
+        }
+
+        public async Task<List<DomainEntity>> FindAllEntitiesAsync()
+        {
+            if (typeof(DbEntity) == typeof(Record))
```

Comment:
Again, doing if statements here and specifying the db model name and the table name defeats the purpose of having this generic method. You can just replace all of this by:
return _mapper.Map<List<DomainEntity>>(await _entities.AsNoTracking().ToListAsync());

and I noticed that Record has a where statement (status != 1). This shouldn't be here, you can have another repo method in the RecordRepository itself to get Records with status != 1.